Source/Controllers/MainViewControllers/TableContent/SPTableContent.m

```diff
@@ -2595,21 +2597,25 @@ - (NSMutableString *)deriveQueryString{
 					   [selectedTable backtickQuotedString], [rowFieldsToSave componentsJoinedAndBacktickQuoted], [rowValuesToSave componentsJoinedByString:@", "]];
 
 	// Otherwise use an UPDATE syntax to save only the changed cells - if this point is reached,
-	// the equality test has failed and so there is always at least one changed cell
+	// the equality test has failed and so there is always at least one changed cell (Except in the case where the cell is of the "generated column" type, the number of cell changed can be 0)
 	} else {
-		queryString = [NSMutableString stringWithFormat:@"UPDATE %@ SET ", [selectedTable backtickQuotedString]];
-		for (i = 0; i < [rowFieldsToSave count]; i++) {
-			if (i) [queryString appendString:@", "];
-			[queryString appendFormat:@"%@ = %@",
-									   [[rowFieldsToSave safeObjectAtIndex:i] backtickQuotedString], [rowValuesToSave safeObjectAtIndex:i]];
-		}
-		NSString *whereArg = [self argumentForRow:-2];
-		if(![whereArg length]) {
-			SPLog(@"Did not find plausible WHERE condition for UPDATE.");
-			NSBeep();
-			return (NSMutableString*)@"";
-		}
-		[queryString appendFormat:@" WHERE %@", whereArg];
+        if ([rowFieldsToSave count] > 0) {
```

Comment:
I think you can combine this else with an if inside of it to an `else if`

Reply:
Actually, even better:
The early return paradigm is a concept that the healthy/normal flow of code should go from the beginning to the end of the method, with early returns for breakout/failing conditions.

In this case, that paradigm would say that you should check if `[rowFieldsToSave count] == 0` in which case you should return that empty string right away and then not have the else condition because the rest of the block wouldn't need to be nested. Does that make any sense?

https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
Yes, it makes sense. I change the code.
But, there is something about that method I don't like. Usually I don't use `return` inside a method but only at the end for readability.